app/services/matched_content_change_generation_service.rb

```diff
@@ -8,6 +8,10 @@ def self.call(*args)
   end
 
   def call
+    # if we already have any records we assume this process was completed
+    # previously and doesn't need to run again.
+    return if MatchedContentChange.exists?(content_change: content_change)
```

Comment:
It looks like the `import!` operation isn't atomic - [not least because it maps over SubscriberLists](https://github.com/alphagov/email-alert-api/blob/master/app/services/matched_content_change_generation_service.rb#L26). Is there a risk here that we could abort `import!` halfway through, and miss out a load of subscribers on a re-run?

Reply:
I believe it is atomic. It does the map before trying to insert and then inserts all of those records in a single query so it either all succeeds or doesn't.

Reply:
Ah, OK. I see now from the docs that it should be a single `insert` statement, since we haven't set a `batch_size`. Could we tweak the comment slightly to explain? (The "assume" part was making me a little nervous.)

Reply:
Yup sure, I've got another small tweak to make too. Will ping you when it's up.

Reply:
I've changed the copy. I decided against my other change as being a bit too edge case.